src/Sylius/Bundle/WebBundle/Resources/translations/menu.en.yml

```diff
@@ -58,6 +58,7 @@ sylius:
                 archetypes: Product archetypes
                 assortment: Assortment
                 attributes: Configure attributes
+                association_type: Association Types
```

Comment:
Also the key is wrong, singular instead of plural, see examples below and above.